### PR TITLE
Makes blast doors/shutters and blast door controllers buildable and deconstructible

### PR DESCRIPTION
**Pull request by OperativeLyn**
## About The Pull Request
As the title says, I made blast doors/shutters buildable through the crafting menu (they aren't cheap to build) and blast door controllers through autolathes. As well as making them linkable and deconstructible with a few steps.

**To build shutters**: 10 plasteel, 5 cable coil and airlock electronics (crafting menu)
**To build blast doors**: 15 plasteel, 15 cable coil and airlock electronics (crafting menu)
**To make blast door controllers**: 100 iron and 50 glass (autolathe)
**To link blast doors and blast doors controllers**: Use a multitool on the controller and choose an ID, then use a multitool on the blast door and input the same ID as the controller, put the controller in a button and voilá, it's ready to open and close.
**To deconstruct blast doors/shutters**: First you have to open the maintenance panel with a screwdriver, then crowbar the electronics out, cut the cables with a wirecutter and finally tear it down with a welder. Each step takes some time so it isn't as easy to deconstruct. NOTE: You can only deconstruct it if its already open. You cant open the panel while its closed.

## Why It's Good For The Game
I love building stuff in ss13 and I think this is a great addition to everyone who likes building as I do.

## Changelog
:cl:
add: Added blast doors and shutters to the crafting menu and blast door controllers to autolathes. As well as deconstruction steps.
/:cl:

### CODE REVIEW

**Comment by coiax** [comment]
Question, why are we using the crafting menu to crate these, when airlocks, firelocks are made using frames out of metal, and then slowly constructioned together with wrenches and screwdrivers and electronics?

**Comment by HugoOdaX** [comment]
> Question, why are we using the crafting menu to crate these, when airlocks, firelocks are made using frames out of metal, and then slowly constructioned together with wrenches and screwdrivers and electronics?

Didn't think about that, but I think crafting menu should be ok, it takes a bit of time so you cant make it in like 5 seconds

**Comment by Jimmius** [comment]
This is cool, but making them deconstructible sort of removes the power of shutters, which is that they almost totally lock down an area, usually requiring you to painstakingly take down a plasteel wall to bypass (Or trying to smash down their absurd integrity and damage reflect). Only a few deconstruction steps requiring such basic tools as a crowbar, screwdriver, and welder takes that out of them. That's less steps than a bloody firelock.

**Comment by HugoOdaX** [comment]
> This is cool, but making them deconstructible sort of removes the power of shutters, which is that they almost totally lock down an area, usually requiring you to painstakingly take down a plasteel wall to bypass (Or trying to smash down their absurd integrity and damage reflect). Only a few deconstruction steps requiring such basic tools as a crowbar, screwdriver, and welder takes that out of them. That's less steps than a bloody firelock.

Well that's why I made it so you can only do it if they are open. You cant deconstruct them if they are closed which doesnt allow people to cheese it and open them

**Comment by Low3** [comment]
Is it possible to ID lock the button so only specific access can use them? (_so i can build mime and curator only autism fort in library_)

**Comment by HugoOdaX** [comment]
> Is it possible to ID lock the button so only specific access can use them? (_so i can build mime and curator only autism fort in library_)

yup, just put in the same button airlock electronics with your access

**Comment by Low3** [comment]
Epic

File: code/game/machinery/doors/poddoor.dm [hunk 1]
```diff
@@ -15,6 +15,53 @@
 	resistance_flags = FIRE_PROOF
 	damage_deflection = 70
 	poddoor = TRUE
+	var/ertblast = FALSE
+	var/deconstruction = INTACT
```
Comment (Jared-Fogle) [inline code/game/machinery/doors/poddoor.dm:19]: Document these.

File: code/game/machinery/doors/poddoor.dm [hunk 2]
```diff
@@ -15,6 +15,53 @@
...
+
+/obj/machinery/door/poddoor/multitool_act(mob/living/user)
+	var/change_id = input("Set the shutters/blast door/blast door controllers ID. It must be a number between 1 and 100.", "ID", id) as num|null
```
Comment (Jared-Fogle) [inline code/game/machinery/doors/poddoor.dm:22]: Does this let you hijack existing shutter doors somehow? Like, can I stop the armory from shutting down just by multitooling the doors?

File: code/modules/assembly/doorcontrol.dm [hunk 3]
```diff
@@ -13,6 +13,12 @@
 	if(id)
 		. += "<span class='notice'>Its channel ID is '[id]'.</span>"
 
+/obj/item/assembly/control/multitool_act(mob/living/user)
```
Comment (Jared-Fogle) [inline code/modules/assembly/doorcontrol.dm:16]: Here too, does this let me just screw with the shutter doors in security just by changing the ID of a completely separated button?

Also, if you're doing this for construction purposes, doesn't being unable to create these defeat the point? Or can you create these already?

Comment (HugoOdaX) [reply code/modules/assembly/doorcontrol.dm:16]: Oh yeah you can create them in an autolathe

Comment (HugoOdaX) [reply code/modules/assembly/doorcontrol.dm:16]: And yeah you could screw with them, but you can already do that without my pr, just screwdrive the button and take them out

File: code/game/machinery/doors/poddoor.dm (hunk 2 above)
Comment (HugoOdaX) [reply code/game/machinery/doors/poddoor.dm:22]: Yep, I should probably edit it so you can only change it if the panel is open, my bad

**Comment by Tlaltecuhtli** [comment]
seems to expensive, drop it to 10 plasteel like a vault door?

File: code/game/machinery/doors/poddoor.dm (hunk 2 above)
Comment (HugoOdaX) [reply code/game/machinery/doors/poddoor.dm:22]: its done

**Comment by HugoOdaX** [comment]
I found the silly space, it's ready now

**Comment by KathrinBailey** [comment]
Keep in mind that I have previously nerfed shutters to be much less durable (they used to inherit their armour block from blast doors), so you can make them cheaper if you want. Removing them by force is much easier than before.
Blast doors are still a giant pain though, them being expensive is just fine considering how easy it is to grief with them. You can kinda just build them in hallways, and they're never getting removed if they get closed.

**Comment by Jared-Fogle** [comment]
Can you confirm this is tested? Web edit PRs are being restricted now, and I didn't notice until now.

**Comment by HugoOdaX** [comment]
> Can you confirm this is tested? Web edit PRs are being restricted now, and I didn't notice until now.

Yup, I tested everything, I did the changes on web because they were minor but I tested them

**Comment by Jared-Fogle** [comment]
> Yup, I tested everything, I did the changes on web because they were minor but I tested them

For future reference, this is a new addition in the contributing guidelines:

> You are expected to have tested your pull requests if it is anything that would warrant testing. **Text only changes, single number balance changes, and similar generally don't need testing, but anything else does. This means by extension web edits are disallowed for larger changes.**

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-5). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Comment by redmoogle** [comment]
https://github.com/yogstation13/Yogstation/pull/10548

_no credit smh_

**Pull request merged**